### PR TITLE
Use ruff instead of pylint, black, isort, flake8 and bandit

### PR DESCRIPTION
**Pull request by bieniu**
This PR:
- migrates pylint, black, isort, flake8 and bandit to ruff
- improves typing
- improves code in several places to meet ruff's requirements
- updates all pre-commit hooks

### CODE REVIEW

File: aioshelly/block_device/coap.py [hunk 1]
```diff
@@ -229,27 +234,30 @@ def subscribe_updates(
         self.subscriptions[ip_or_device_id] = message_received
         return lambda: self.subscriptions.pop(ip_or_device_id)
 
-    async def __aenter__(self) -> "COAP":
+    async def __aenter__(self) -> Self:
```
Comment (bieniu) [inline aioshelly/block_device/coap.py:237]: ruff complains to use `-> Self`
PYI034 `__aenter__` methods in classes like `COAP` usually return `self` at runtime

File: aioshelly/block_device/coap.py [hunk 2]
```diff
@@ -229,27 +234,30 @@ def subscribe_updates(
...
         """Entering async context manager."""
         await self.initialize()
         return self
 
     async def __aexit__(
-        self, exc_type: Exception, exc_value: str, traceback: TracebackType
+        self,
+        exc_type: type[BaseException] | None,
+        exc_value: BaseException | None,
+        traceback: TracebackType | None,
```
Comment (bieniu) [inline aioshelly/block_device/coap.py:246]: https://docs.astral.sh/ruff/rules/bad-exit-annotation/

File: aioshelly/rpc_device/wsrpc.py [hunk 3]
```diff
@@ -485,7 +494,7 @@ def close(self) -> None:
         """Stop the websocket server."""
         if self._runner is not None:
             loop = asyncio.get_running_loop()
-            loop.create_task(self._runner.cleanup())
+            loop.create_task(self._runner.cleanup())  # noqa: RUF006
```
Comment (thecode) [inline aioshelly/rpc_device/wsrpc.py:497]: Would probably need to address this in the future

File: tools/common/__init__.py [hunk 4]
```diff
@@ -91,20 +92,18 @@ def print_device(device: BlockDevice | RpcDevice) -> None:
...
+        assert device.blocks
 
     for block in device.blocks:
         print(block)
         for attr, value in block.current_values().items():
             info = block.info(attr)
 
             if value is None:
-                value = "-"
+                value = "-"  # noqa: PLW2901
```
Comment (thecode) [inline tools/common/__init__.py:104]: Would be nice to replace it with a temp value (`_value`) and use it for the print
`_value = value if value is not None else "-"`

**Review by thecode — APPROVED** [review]

File: tools/common/__init__.py [hunk 5]
```diff
@@ -104,11 +104,11 @@ def print_block_device(device: BlockDevice) -> None:
             info = block.info(attr)
 
             if value is None:
-                value = "-"  # noqa: PLW2901
+                _value = value if value is not None else "-"
```
Comment (thecode) [inline tools/common/__init__.py:107]: That was my bad, we don't need the first `if` with the 2nd.

Comment (bieniu) [reply tools/common/__init__.py:107]: Oh I'm blind

File: aioshelly/block_device/device.py [hunk 6]
```diff
@@ -16,7 +16,13 @@
 from yarl import URL
 
 from ..common import ConnectionOptions, IpOrOptionsType, get_info, process_ip_or_options
-from ..const import CONNECT_ERRORS, DEVICE_IO_TIMEOUT, HTTP_CALL_TIMEOUT, MODEL_RGBW2
+from ..const import (
+    CONNECT_ERRORS,
+    DEVICE_IO_TIMEOUT,
+    HTTP_CALL_TIMEOUT,
+    HTTP_PORT,
```
Comment (thecode) [inline aioshelly/block_device/device.py:23]:
```suggestion
    DEFAULT_HTTP_PORT,
```

File: aioshelly/const.py [hunk 7]
```diff
@@ -246,5 +246,13 @@
...
-BLOCK_GENERATIONS = (1,)
-RPC_GENERATIONS = (2, 3)
+GEN1 = 1
+GEN2 = 2
+GEN3 = 3
+
+BLOCK_GENERATIONS = (GEN1,)
+RPC_GENERATIONS = (GEN2, GEN3)
+
+HTTP_PORT = 80
```
Comment (thecode) [inline aioshelly/const.py:256]:
```suggestion
DEFAULT_HTTP_PORT = 80
```

File: aioshelly/block_device/device.py [hunk 8]
```diff
@@ -120,8 +126,9 @@ async def initialize(self, async_init: bool = False) -> None:
         if self._initializing:
             raise RuntimeError("Already initializing")
 
-        # GEN1 cannot be configured behind a range extender as CoAP port cannot be natted
-        if self.options.port != 80:
+        # GEN1 cannot be configured behind a range extender as CoAP port cannot be
+        # natted
+        if self.options.port != HTTP_PORT:
```
Comment (thecode) [inline aioshelly/block_device/device.py:131]:
```suggestion
        if self.options.port != DEFAULT_HTTP_PORT:
```

**Review by thecode — APPROVED** [review]

**Review by chemelli74 — APPROVED** [review]
LGTM!

**Pull request merged**